### DISCUSSION TITLE
Get correct count of all members that are not offline

### DISCUSSION THREAD

**Discussion by Steinente** [discordjs/discord.js#9221] [Q&A] [answered]
Hello,

I try to get the number of all members who are not offline.
For this I access the presence status.
The PresenceStatus class provides the following statuses:
'online' | 'idle' | 'dnd' | 'invisible' | 'offline' | undefined

So when I do `if (status === 'online' || status === 'dnd' || status === 'idle' || status === 'invisible') onlineMemberCount++`, unfortunately I don't get the same number as the one Discord calculates on the smartphone, for example, when you select Discord at the top.

I did a little digging and came to this conclusion:
When the bot is started, the members are fetched and the status is queried, the status of all members that have selected 'invisible' is displayed as undefined.
Only when the bot is online and a member changes his status to another and back to 'invisible', the 'invisible' status is displayed and no longer undefined.

Am I missing something? Is there another way to get the exact number? Is this a mistake?

**Comment by sdanialraza** [comment, marked as answer]
Hey, PresenceStatus as stated [here](https://discord.js.org/#/docs/discord.js/main/typedef/PresenceStatus), only provides 4 statuses: `online`, `idle`, `dnd`, and `offline`, and it can also be `null` sometimes. It does not provide `invisible`. For all statuses, you'd just check if it's that value, so for online, you'd do `if (status == "online") onlineMemberCount++`. There's an exception for offline, as if the user is offline, it can either be `null` or `offline`, so you'd do `if (!status | status == "offline") offlineMemberCount++`. Hope this is helpful

**Reply by Steinente** [reply]
![image](https://user-images.githubusercontent.com/42090794/224495030-1a304546-b69e-4a33-a089-b1c0554e0ed0.png)
The `invisible` status exists and works, as described in the origin thread. Unfortunately, just not as hoped for.

**Reply by sdanialraza** [reply]
Ah, I understand what you're trying to do, sorry about that, I thought you were going for something else. The reason this is happening is because of how Discord sends data about invisible users. You can't get everyone who has the status `invisible`. Only when they change to `invisible` when the bot is online. As far as their status goes for you, it'd be `offline` or `null`, not `invisible`. Hope that makes sense. I was actually a bit unsure about how things worked so I asked in [discord.js server](https://discord.com/invite/djs) on Discord and they helped me understand better as well. So I'd recommend asking in the Discord server for a faster response.

**Reply by Steinente** [reply]
Okay, thank you!

**Reply by sdanialraza** [reply]
Of course, you can mark this as the answer so people know this discussion has been answered.
